agent/consul/gateways/bind.go

```diff
@@ -43,8 +51,13 @@ func BindRoutesToGateways(gateways []*gatewayMeta, routes ...structs.BoundRoute)
 				}
 				for _, ref := range references {
 					delete(parentRefs, ref)
+					// this ref successfully bound, add it to the set that we'll update the
+					// status for
+					if _, found := errored[ref]; !found {
+						boundRefs = append(boundRefs, references...)
+					}
 				}
-			} else if gateway.unbindRoute(route) {
+			} else if gateway.unbindRoute(routeRef) {
```

Comment:
Switched to using refs directly for binding/unbinding so that we can unbind even when we no longer have a full object because of something like a route being deleted that we need to remove from our bound gateway.